lib/reek/configuration/app_configuration.rb

```diff
@@ -5,89 +5,115 @@ module Reek
   # @api private
   module Configuration
     # @api private
-    class ConfigFileException < StandardError; end
     #
     # Reek's singleton configuration instance.
     #
     # @api private
-    module AppConfiguration
-      NON_SMELL_TYPE_KEYS   = %w(exclude_paths)
-      EXCLUDE_PATHS_KEY     = 'exclude_paths'
-      @configuration        = {}
-      @has_been_initialized = false
+    class AppConfiguration
+      EXCLUDE_PATHS_KEY = 'exclude_paths'
+      attr_reader :exclude_paths, :default_directive, :directory_directives
 
-      class << self
-        attr_reader :configuration
+      # Given this configuration file:
+      #
+      # ---
+      # IrresponsibleModule:
+      #   enabled: false
+      # "app/helpers":
+      #   UtilityFunction:
+      #     enabled: false
+      # exclude_paths:
+      #   - "app/controllers"
+      #
+      # this would result in the following configuration:
+      #
+      # exclude_paths = [Pathname('spec/samples/two_smelly_files')]
+      # default_directive = { Reek::Smells::IrresponsibleModule=>{"enabled"=>false} }
+      # directory_directives = { Pathname("spec/samples/three_clean_files/") =>
+      #                          {Reek::Smells::UtilityFunction=>{"enabled"=>false}}}
```

Comment:
I know it borders on minutiæ, but can we either keep or drop spaces inside the braces and around the hashrockets here?

(I understand this was copy-’n’-pasted from irb, but I’d rather have a RuboCop-kosher syntax with spaces – or at least not have them everywhere rather than this mish-mash.)